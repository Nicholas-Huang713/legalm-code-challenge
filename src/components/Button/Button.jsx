/**
 * @module Button
 * @description A simple Button component that handles action when clicked.
 * @param {object} props - The component props.
 * @param {string} props.btnText - Text to display in button.
 * @param {function} props.handleClick - Function to be called with button is clicked.
 * @param {string} props.value - Value of button.
 */

export default function Button({ btnText, handleClick, value }) {
  return (
    <button value={value ? value : null} onClick={handleClick}>
      {btnText}
    </button>
  );
}
