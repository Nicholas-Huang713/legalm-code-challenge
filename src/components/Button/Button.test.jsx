import { render, fireEvent, screen } from "@testing-library/react";
import Button from "./Button";

const onClickMock = vi.fn();
const renderButton = () =>
  render(<Button btnText="Hello" handleClick={onClickMock} value="Hello" />);

describe("Button component", () => {
  it("renders button component", () => {
    const { container, getByText } = renderButton();
    expect(getByText("Hello")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it("onClick method called", () => {
    renderButton();
    const btn = screen.getByText("Hello");
    fireEvent.click(btn);
    expect(onClickMock).toHaveBeenCalledTimes(1);
  });
});
