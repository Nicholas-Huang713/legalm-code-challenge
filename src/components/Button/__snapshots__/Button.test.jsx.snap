// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button component > renders button component 1`] = `
<div>
  <button
    value="Hello"
  >
    Hello
  </button>
</div>
`;
